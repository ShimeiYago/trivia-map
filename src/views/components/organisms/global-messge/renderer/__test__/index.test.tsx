import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, State, Props } from '..';

let shallowWrapper: ShallowWrapper<Props, State, Renderer>;

const props: Props = {
  articleFormFetchingState: 'waiting',
  articleFormSubmittingState: 'waiting',
  readingArticleLoadingState: 'waiting',
  closeArticleModal: jest.fn(),
  closeFormModal: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<Renderer {...props} />);
  });

  it('basic', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('componentDidUpdate', () => {
  it('should change states to show post success message', () => {
    shallowWrapper = shallow(<Renderer {...props} />);
    shallowWrapper.setProps({
      articleFormSubmittingState: 'success',
    });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate({
      ...props,
      articleFormSubmittingState: 'loading',
    });
    expect(instance.state.message).toBe('投稿が完了しました！');
  });

  it('should change states to show fetching article error message', () => {
    shallowWrapper = shallow(<Renderer {...props} />);
    shallowWrapper.setProps({
      readingArticleLoadingState: 'error',
      readingArticleErrorMsg: 'fetch error',
    });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate({
      ...props,
      readingArticleLoadingState: 'loading',
    });
    expect(instance.state.message).toBe('fetch error');
  });

  it('should change states to show fetching article empty error message', () => {
    shallowWrapper = shallow(<Renderer {...props} />);
    shallowWrapper.setProps({
      readingArticleLoadingState: 'error',
    });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate({
      ...props,
      readingArticleLoadingState: 'loading',
    });
    expect(instance.state.message).toBe('');
  });

  it('should change states to show fetching form error message', () => {
    shallowWrapper = shallow(<Renderer {...props} />);
    shallowWrapper.setProps({
      articleFormFetchingState: 'error',
      articleFormFetchingErrorMsg: 'fetch error',
    });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate({
      ...props,
      articleFormFetchingState: 'loading',
    });
    expect(instance.state.message).toBe('fetch error');
  });

  it('should change states to show fetching empty form error message', () => {
    shallowWrapper = shallow(<Renderer {...props} />);
    shallowWrapper.setProps({
      articleFormFetchingState: 'error',
    });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate({
      ...props,
      articleFormFetchingState: 'loading',
    });
    expect(instance.state.message).toBe('');
  });
});

describe('handleCloseMessage', () => {
  it('should change show state', () => {
    shallowWrapper = shallow(<Renderer {...props} />);
    shallowWrapper.setState({ show: true });
    const instance = shallowWrapper.instance();

    instance['handleCloseMessage']();
    expect(instance.state.show).toBeFalsy();
  });
});
