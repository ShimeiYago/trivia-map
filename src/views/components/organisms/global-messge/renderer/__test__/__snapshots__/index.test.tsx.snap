// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<ForwardRef(Snackbar)
  autoHideDuration={6000}
  onClose={[Function]}
  open={false}
>
  <ForwardRef(Alert)
    onClose={[Function]}
    severity="success"
    sx={
      Object {
        "width": "100%",
      }
    }
    variant="filled"
  >
    <ForwardRef(Typography)
      fontSize="inherit"
    />
    <NonStyleLink
      to="/articles/1"
    >
      <ForwardRef(Typography)
        fontSize="inherit"
        sx={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        投稿内容を確認する
      </ForwardRef(Typography)>
    </NonStyleLink>
  </ForwardRef(Alert)>
</ForwardRef(Snackbar)>
`;
